### PR TITLE
8330621: Make 5 compiler tests use ProcessTools.executeProcess

### PR DESCRIPTION
**Pull request by lepestock**
Said tests use simple `new ProcessBuilder` and its `start` method to start secondary processes.

As stated in [JDK-8174768](https://bugs.openjdk.org/browse/JDK-8174768), we try to have more information about started secondary processes and make the execution more controllable. This PR makes those tests use ProcessTools.executeProcess instead of using the `.start` method.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8330621](https://bugs.openjdk.org/browse/JDK-8330621): Make 5 compiler tests use ProcessTools.executeProcess (**Enhancement** - P4)

### Reviewers
 * [Christian Hagedorn](https://openjdk.org/census#chagedorn) (@chhagedorn - **Reviewer**) ⚠️ Review applies to [c4641d05](https://git.openjdk.org/jdk/pull/18856/files/c4641d054a9e88008306647326860eb929f10eac)
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18856/head:pull/18856` \
`$ git checkout pull/18856`

Update a local copy of the PR: \
`$ git checkout pull/18856` \
`$ git pull https://git.openjdk.org/jdk.git pull/18856/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18856`

View PR using the GUI difftool: \
`$ git pr show -t 18856`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18856.diff">https://git.openjdk.org/jdk/pull/18856.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18856#issuecomment-2066266107)

### CODE REVIEW

File: test/hotspot/jtreg/compiler/profiling/spectrapredefineclass_classloaders/Launcher.java [hunk 1]
```diff
@@ -65,7 +66,7 @@ public static void main(String[] args) throws Exception  {
 
         ProcessBuilder pb = new ProcessBuilder(jar.getCommand());
         try {
-            OutputAnalyzer output = new OutputAnalyzer(pb.start());
+            OutputAnalyzer output = ProcessTools.executeProcess(pb);
             output.shouldHaveExitValue(0);
         } catch (IOException ex) {
```
Comment (chhagedorn) [inline test/hotspot/jtreg/compiler/profiling/spectrapredefineclass_classloaders/Launcher.java:71]: Can probably also be updated as in Test7068051?
```suggestion
        } catch (Exception ex) {
```

**Review by chhagedorn — APPROVED** [review]
Looks good!

**Review by stefank — APPROVED** [review]
Looks good to me.

Do you know that you also could skip explicitly creating ProcessBuilders and instead run:
```
ProcessTools.executeProcess(jar.getCommand());
```

**Comment by lepestock** [comment]
> Do you know that you also could skip explicitly creating ProcessBuilders and instead run:
Tried to be as less intrusive as possible :). But I agree, the fewer lines the better. Fixed.

**Review by stefank — APPROVED** [review]

**Review by TobiHartmann — APPROVED** [review]

**Pull request closed without merging**